File: backend/app/api/leads.py
from flask import jsonify, request
from app import db
from app.models import Lead
from app.api import bp

@bp.route('/leads', methods=['POST'])
def create_lead():
    data = request.get_json()
    
    lead = Lead(
        name=data['name'],
        mobile=data['mobile'],
        email=data.get('email'),
        vehicle_type=data['vehicleType'],
        state=data['state'],
        service_type=data['serviceType'],
        comments=data.get('comments')
    )
    
    db.session.add(lead)
    db.session.commit()
    
    return jsonify(lead.to_dict()), 201

@bp.route('/leads', methods=['GET'])
def get_leads():
    leads = Lead.query.order_by(Lead.created_at.desc()).all()
    return jsonify([lead.to_dict() for lead in leads])

File: src/api/leads.ts
import axios from 'axios';

const API_URL = 'http://localhost:5000/api';

export interface LeadData {
  name: string;
  mobile: string;
  email?: string;
  vehicleType: string;
  state: string;
  serviceType: string;
  comments?: string;
}

export const createLead = async (leadData: LeadData) => {
  try {
    const response = await axios.post(`${API_URL}/leads`, leadData);
    return response.data;
  } catch (error) {
    throw new Error('Failed to submit lead');
  }
};